packages/client/src/helpers/runWithRetry.ts

```diff
@@ -0,0 +1,87 @@
+const sleep = (time: number) =>
+  new Promise<void>((resolve) => {
+    setTimeout(resolve, time);
+  });
```

Comment:
nit-pick, but I think there is a `sleep` function already available in `coordinator/connection/utils.ts`